Source/objects.cpp

```diff
@@ -2437,352 +2437,345 @@ void RedoPlayerVision()
 	}
 }
 
+/**
+ * @brief Checks if an open door can be closed
+ *
+ * In order to be able to close a door the space where the closed door would be must be free of bodies, monsters, and items
+ *
+ * @param doorPos Map tile where the door is in its closed position
+ * @return true if the door is free to be closed, false if anything is blocking it
+*/
+static inline bool isDoorClear(const Point &doorPosition)
+{
+	return dDead[doorPosition.x][doorPosition.y] == 0
+		&& dMonster[doorPosition.x][doorPosition.y] == 0
+		&& dItem[doorPosition.x][doorPosition.y] == 0;
+}
+
 void OperateL1RDoor(int pnum, int oi, bool sendflag)
 {
-	int xp, yp;
+	ObjectStruct &door = object[oi]; // Non const ref as flags are updated in this function

```

Comment:
Idk, this comment seams pretty obvious, how would you open a door by only inspecting it?
```suggestion
	ObjectStruct &door = object[oi];
```